lib/ui/text/line_metrics.h

```diff
@@ -0,0 +1,93 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_LIB_UI_TEXT_LINE_METRICS_H_
+#define FLUTTER_LIB_UI_TEXT_LINE_METRICS_H_
+
+#include "flutter/lib/ui/text/run_metrics.h"
+#include "flutter/third_party/txt/src/txt/run_metrics.h"
+#include "third_party/dart/runtime/include/dart_api.h"
+#include "third_party/tonic/converter/dart_converter.h"
+
+namespace flutter {
+
+struct LineMetrics {
```

Comment:
Docs? For example, it would be interesting to know why the memory layout is chosen to be flat.